topostats/tracing/splining.py

```diff
@@ -587,13 +591,14 @@ def splining_image(
                 }
                 molstats[grain_no.split("_")[-1] + "_" + mol_no.split("_")[-1]] = {
                     "image": filename,
-                    "grain_number": grain_no.split("_")[-1],
+                    "grain_number": int(grain_no.rsplit("_", maxsplit=1)[-1]),
+                    "molecule_number": int(mol_no.rsplit("_", maxsplit=1)[-1]),
                 }
                 molstats[grain_no.split("_")[-1] + "_" + mol_no.split("_")[-1]].update(tracing_stats)
                 LOGGER.info(f"[{filename}] : Finished splining {grain_no} - {mol_no}")
 
             except Exception as e:  # pylint: disable=broad-exception-caught
```

Comment:
Broad exceptions should be avoided if possible.

Are there any specific errors that have arisen during processing that we could explicitly list here?

Reply:
Not that I know of, but it was meant to be a catch-all to not stop all the TS processing. 

I know it's bad practice but I think it's more detrimental if an unknown-to-us exception of prevents the processing of a users whole dataset and they don't have the coding knowledge to read the error and add the specific exception themselves. This way, if they have an issue with the processing not capturing the molecule(s), the error is still output in the logs should they want to bring it to the TS forums to help diagnose the issue, or decide that the one file/molecule isn't something to be worried about.

So a user experience vs best practice debate. What do you think?

Reply:
> I know it's bad practice but I think it's more detrimental if an unknown-to-us exception of prevents the processing of a users whole dataset and they don't have the coding knowledge to read the error and add the specific exception themselves. This way, if they have an issue with the processing not capturing the molecule(s), the error is still output in the logs **should they want to bring it to the TS forums to help diagnose the issue, or decide that the one file/molecule isn't something to be worried about.**


Emphasis on what I think the key point is. We don't see a huge amount of reports about errors really. @TFirth2 highlighted one of the problems with having a broad `except` elsewhere in the codebase in #887 although in this instance the offending file is returned in the log. Beyond that we rarely see reports of problematic images stopping processing.

I'm not sure how many actual reports of such errors we would see if processing doesn't halt. I get the impression there are usually quite a few pictures being processed (batch processing is after all one of the original purposes of TopoStats) and so one not processing will likely just be ignored and we won't see a report of the problem logged. In turn we're less likely to find out what errors can arise and then never be able to improve and handle them.

> So a user experience vs best practice debate. What do you think?

More an issue of how to improve the software and find out about images that are problematic from my perspective.

Reply:
Interestingly #887 was created _because_ it stopped processing the dataset and the want was to know which image caused the error so it could be removed and the dataset run until the end.

Although I do see your point and concerns about not knowing about flaws in the code and employing the ostrich algorithm.

Could instead, we go through and change logging levels for `error` / `debug` statements for example, and by default use `info` to produce a reduced logging output and thus be more obvious to users that things have gone wrong and where?
I know when I first started I just used `logger.info` for everything as anything else was at one point broken but seems like "warning" and "error" and "debug" now work. 

I'll sleep on it tonight and have a chat with others tomorrow and gather their opinions too.

Reply:
The deadline is tight here. I in principle agree that we should be letting things crash, which will lead to less tech debt and more stable software later on, however we need this merging ASAP for the publication to go ahead.

I personally would like to overhaul error handling a little in topostats, and would like to make an issue about it. Would it be okay if this be handled there instead of here?

Reply:
Added these in commit [9a73090](https://github.com/AFM-SPM/TopoStats/pull/898/commits/9a7309013f1d114f105c0b09a7aad687179d9d40) and issue